synapse_rock/rockcraft.yaml

```diff
@@ -183,3 +187,51 @@ parts:
     source: attributemaps
     organize:
       login_ubuntu.py: /usr/local/attributemaps/login_ubuntu.py
+  mas-templates:
+    plugin: dump
+    <<: *mas-source
+    source-subdir: templates
+    organize:
+      "*": mas/share/templates/
+  mas-translations:
+    plugin: dump
+    <<: *mas-source
+    source-subdir: translations
+    organize:
+      "*": mas/share/translations/
+  mas-assets:
+    plugin: nil
+    <<: *mas-source
+    source-subdir: frontend
+    build-environment:
+      - NODE_URI: "https://nodejs.org/dist/v22.11.0/node-v22.11.0-linux-x64.tar.gz"
+      - NODE_ENV: dev
+    override-build: |
+      curl -Ls $NODE_URI | tar xzf - -C /usr/ --skip-old-files --no-same-owner --strip-components=1
+      npm --prefix=$CRAFT_PART_BUILD/frontend ci
+      npm --prefix=$CRAFT_PART_BUILD/frontend run build
+      mkdir -p $CRAFT_PART_INSTALL/mas/share/assets
+      cp -r frontend/dist/* $CRAFT_PART_INSTALL/mas/share/assets/
+      cp frontend/dist/manifest.json $CRAFT_PART_INSTALL/mas/share/manifest.json
+    stage:
+      - mas/*
+  mas-cli:
+    plugin: rust
+    rust-channel: stable
+    rust-cargo-parameters:
+      - "--features dist"
+      - "--no-default-features"
+    <<: *mas-source
+    override-build: |
+      curl -L -o opa https://openpolicyagent.org/downloads/v0.70.0/opa_linux_amd64_static
```

Comment:
I'll make a follow-up PR to build the opa from source / verify the binary checksum